fs_test.go

```diff
@@ -44,6 +45,186 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+func TestSeeker(t *testing.T) {
+	s3cl := newClient(t)
+
+	const testFile = "file.txt"
+	content := []byte("content")
+
+	createBucket(t, s3cl, *bucket)
+	cleanBucket(t, s3cl, *bucket)
+
+	writeFile(t, s3cl, *bucket, testFile, content)
+
+	t.Cleanup(func() {
+		cleanBucket(t, s3cl, *bucket)
+
+		t.Log("test stats:")
+		t.Log("ListObjects calls:", atomic.LoadInt64(&listC))
+		t.Log("GetObject calls:", atomic.LoadInt64(&getC))
+	})
+
+	t.Run("seek once", func(t *testing.T) {
+		fixtures := []struct {
+			desc     string
+			offset   int64
+			whence   int
+			expected int64
+			err      error
+		}{
+			{"whence SeekStart ", 2, io.SeekStart, 2, nil},
+			{"whence SeekCurrent", 4, io.SeekCurrent, 4, nil},
+			{"whence SeekEnd", -1, io.SeekEnd, int64(len(content)) - 1, nil},
+			{"whence SeekEnd", 4, 3, 0, fmt.Errorf("unknown 'whence': 3")},
```

Comment:
this needs to be updated, please use `errors.New()`

I am not a fan of comparing errors by their message... but I guess it's okay for something as simple

Reply:
I'm still comparing errors by their message, but those tests are in their own test-subgroup now (starting from line 115).